dynamiqs/qarrays/dense_qarray.py

```diff
@@ -94,6 +98,10 @@ def squeeze(self, axis: int | tuple[int, ...] | None = None) -> QArray | Array:
         else:
             return DenseQArray(self.dims, data)
 
+    def eig(self) -> tuple[Array, QArray]:
```

Comment:
I think we should have this as private for now, this will cause endless troubles with superoperators for the same reason we had trouble proposing a proper `dq.eig()` method.